app/controllers/team_members_controller.rb

```diff
@@ -1,42 +1,50 @@
 class TeamMembersController < ApplicationController
 
-  def edit
-    @team_member = TeamMember.find(params[:id])
+  before_action :set_team_member
 
+  def edit
     if @team_member.role == 'custodian'
       @team_members = @team_member.team.team_members.joins(:user).where.not(users: {full_name: ''} )
     end
   end
 
   def update
-
-    team_member = TeamMember.find(params[:id])
-
-    if team_member.role == 'custodian'
+    if @team_member.role == 'custodian'
       if params[:new_custodian_team_member_id] == params[:id]
         flash[:notice] = 'Your custodian remains unchanged'
       else
         new_custodian = TeamMember.find(params[:new_custodian_team_member_id])
 
-        team_member.role = 'advanced'
+        @team_member.role = 'advanced'
         new_custodian.role = 'custodian'
 
-        team_member.save
+        @team_member.save
         new_custodian.save
 
         flash[:notice] = 'The custodian has been changed'
       end
-      redirect_to team_path(team_member.team_id)
+      redirect_to team_path(@team_member.team_id)
     else
-      if team_member.update(role: params[:team_member][:role])
+      if @team_member.update(role: params[:team_member][:role])
         flash[:notice] = 'Your update has been successful'
-        redirect_to team_path(team_member.team_id)
+        redirect_to team_path(@team_member.team_id)
       else
         flash[:notice] = 'Your update has failed'
         render :edit
       end
     end
+  end
+
+  def destroy
+    @team_member.destroy
+    flash[:notice] = "Successfully removed #{@team_member.user.full_name} from team"
+    redirect_to team_path(@team_member.team_id)
+  end
 
+  private
+
+  def set_team_member
```

Comment:
Well rails g scaffold foobar generates all of the routes, when you have all of the routes you have edit, show, destroy + that requires @foobar. Thats 3+ mentions meaning that its good code to do it.

Otherwise if you really wanna keep it, use the `only:` annotation to say the methods so new methods in the file dont automatically inherit it.